modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/dtx/DtxBranch.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2016, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.andes.kernel.dtx;
+
+import org.apache.log4j.Logger;
+import org.wso2.andes.kernel.AndesAckData;
+import org.wso2.andes.kernel.AndesChannel;
+import org.wso2.andes.kernel.AndesException;
+import org.wso2.andes.kernel.AndesMessage;
+import org.wso2.andes.kernel.disruptor.inbound.AndesInboundStateEvent;
+import org.wso2.andes.kernel.disruptor.inbound.InboundEventManager;
+import org.wso2.andes.kernel.slot.SlotMessageCounter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.transaction.xa.Xid;
+
+public class DtxBranch implements AndesInboundStateEvent {
+
+    /**
+     * Class logger
+     */
+    private static final Logger LOGGER = Logger.getLogger(DtxBranch.class);
+
+    /**
+     * Internal XID has this value when the branch is not in prepared state
+     */
+    public static final int NULL_XID = -1;
+
+    private final Xid xid;
+    private final DtxRegistry dtxRegistry;
+    private Map<Long, State> associatedSessions = new HashMap<>();
+    private InboundEventManager eventManager;
+    private Runnable callback;
+
+    /**
+     * Getter for enqueueList
+     */
+    public ArrayList<AndesMessage> getEnqueueList() {
+        return enqueueList;
+    }
+
+    /**
+     * Keep the list of messages that need to be published when the transaction commits
+     */
+    private ArrayList<AndesMessage> enqueueList = new ArrayList<>();
+
+    /**
+     * Getter for dequeueList
+     */
+    public List<AndesAckData> getDequeueList() {
+        return dequeueList;
+    }
+
+    /**
+     * Keep the list of messages that need to be acked when the transaction commits
+     */
+    private List<AndesAckData> dequeueList = new ArrayList<>();
+
+    /**
+     * Current branch state
+     */
+    private State state = State.ACTIVE;
+
+    /**
+     * Used to keep the internal xid value used in message store
+     */
+    private long internalXid = NULL_XID;
```

Comment:
Fixed in https://github.com/wso2/andes/pull/768/commits/c7b9fdb21ea17a2f3a95ee41c7bcef2308881efb